### PR TITLE
Deprecated /(un)follow commands and respective usercard action

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

As announced, `/helix/users/follows` endpoints have been decomissioned today.

farewell, beautiful Helix code

CC #3076 #2629

### CODE REVIEW

**Comment by zneix** [comment]
For now, `/follow` and `/unfollow` have been marked as deprecated:
saj

And for the time being, follow checkbox in usercard remains but it's permanently disabled:
CoolCat

This maybe isn't the ideal solution as it could confuse users, making them think they can still (un)follow and try to click the checkbox.
We could either:
1. Remove the checkbox and checking if current user follows the user of whom usercard is opened altogether.
2. Think of some alternative, like the one that @mm2pl suggested:

<details>
<summary>Click to expand</summary>

GIMP masters

</details>

File: src/controllers/commands/CommandController.cpp [hunk 1]
```diff
@@ -373,6 +373,22 @@ void CommandController::initialize(Settings &, Paths &paths)
             return "";
         });
 
+    this->registerCommand("/follow", [](const auto &words, auto channel) {
+        channel->addMessage(makeSystemMessage(
+            "Twitch has removed possibility to follow users through "
```
Comment (Felanbird) [inline src/controllers/commands/CommandController.cpp:378]:
```suggestion
            "Twitch has removed the ability to follow users through "
```

File: src/controllers/commands/CommandController.cpp [hunk 2]
```diff
@@ -373,6 +373,22 @@ void CommandController::initialize(Settings &, Paths &paths)
...
+            "third-party applications. For more information, see "
+            "https://github.com/Chatterino/chatterino2/issues/3076"));
+        return "";
+    });
+
+    this->registerCommand("/unfollow", [](const auto &words, auto channel) {
+        channel->addMessage(makeSystemMessage(
+            "Twitch has removed possibility to unfollow users through "
```
Comment (Felanbird) [inline src/controllers/commands/CommandController.cpp:386]:
```suggestion
            "Twitch has removed the ability to unfollow users through "
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## Unversioned
 
 - Major: Newly uploaded Twitch emotes are once again present in emote picker and can be autocompleted with Tab as well. (#2992)
+- Major: Deprecated `/(un)follow` commands and (un)following in the usercards as Twitch removed the possibility to (un)follow users.
```
Comment (Felanbird) [inline CHANGELOG.md:6]:
```suggestion
- Major: Deprecated `/(un)follow` commands and (un)following in the usercards as Twitch has removed this feature for 3rd party applications. (#3076, #3078)
```

**Review by Felanbird — CHANGES_REQUESTED** [review]
`Minor English loophole:`
Twitch has taken away our ability to manage follows, not the quality of, or possibility to make changes for the endpoint. 
https://wikidiff.com/ability/possibility

As for the changelog I'm indifferent in the usage of (un)follow 🤷

**Comment by Felanbird** [comment]
As for the future of the button, I'm going to put my vote for keeping the button with its ability show follow status as currently handled, but making the functionality return an error message, as done with the slash commands in this PR.

Possibly removing it in the future if we need button room for some reason.

**Comment by zneix** [comment]
> but making the functionality return an error message, as done with the slash commands in this PR.

As I've stated in my comment under this PR:

> follow checkbox in usercard remains but it's permanently disabled

Do you mean we should keep the functionality of the button but add error messages notifying users that (un)following failed or did you just overlook that part where I said it? I believe there should be no logic for event actions related to that button as it's gonna be literally cosmetic at this point (if we decide to keep it in the first place that is).

**Comment by Felanbird** [comment]
> Do you mean we should keep the functionality of the button but add error messages notifying users that (un)following failed 

Yes, I think if we keep the button it should at least do something indicating to users that the attempt failed.




> I believe there should be no logic for event actions related to that button as it's gonna be literally cosmetic at this point  (if we decide to keep it in the first place that is).

If we to remove the all logic we should just remove the button in its entirety, no need for a cosmetic button.

**Review by pajlada — APPROVED** [review]

**Comment by Felanbird** [comment]
Works as intended, the only problem seems to be the loading time, you have hover the button for 1.5~ seconds before the prompt appears. It took me a while to actually get the prompt because I was clicking the "button" and and doing so resets the loading.

image

**Comment by pajlada** [comment]
> Works as intended, the only problem seems to be the loading time, you have hover the button for 1.5~ seconds before the prompt appears. It took me a while to actually get the prompt because I was clicking the "button" and and doing so resets the loading.
> 
> image

That's unfortunately fairly tricky for us to solve without creating a new QCheckBox type and re-implementing it's click handler making it show the tooltip.

**Comment by Mm2PL** [comment]
About the usercard: maybe we could change the text color/effects. For example gray out the `Follow` text or cross it out.

**Comment by pajlada** [comment]
All in favour of removing the checkbox?

**Review by Felanbird — APPROVED** [review]

**Review by pajlada — APPROVED** [review]

**Review by Mm2PL — APPROVED** [review]
lgtm

**Pull request merged**